src/execute/oracle_approval.rs

```diff
@@ -1,113 +1,135 @@
+            }
+            Err(_) => {
+                return ContractError::PayableNotFound {
+                    payable_uuid: oracle_approval.payable_uuid,
+                }
+                .to_result();
+            }
+        };
     // Only the designated oracle can mark an approval on a denomination
-    if info.sender != state.oracle_address {
+    if info.sender != scope_attribute.oracle_address {
         return Err(ContractError::Unauthorized);
     }
-    let mut payables_bucket = payable_meta_storage(deps.storage);
-    // Ensure the target payable definition actually exists
-    let mut target_payable = match payables_bucket.load(oracle_approval.payable_uuid.as_bytes()) {
-        Ok(meta) => {
-            if meta.oracle_approved {
-                return Err(ContractError::DuplicateApproval {
-                    payable_uuid: oracle_approval.payable_uuid,
-                });
-            }
-            meta
-        }
-        Err(_) => {
-            return Err(ContractError::PayableNotFound {
-                payable_uuid: oracle_approval.payable_uuid,
-            });
-        }
-    };
-    // TODO: Tag an attribute on the scope once functionality is available
+    let state = config_read_v2(deps.storage).load()?;
     // The oracle is paid X on each approval, where X is the remaining amount after the fee is taken
     // from the onboarding funds.
     let oracle_withdraw_amount =
         state.onboarding_cost - state.onboarding_cost.mul(state.fee_percent);
     // Only create a payment to the oracle if there were funds stored in the first place
     if oracle_withdraw_amount.u128() > 0 {
         messages.push(CosmosMsg::Bank(BankMsg::Send {
-            to_address: state.oracle_address.into(),
+            to_address: scope_attribute.oracle_address.clone().into(),
             amount: vec![coin(oracle_withdraw_amount.u128(), state.onboarding_denom)],
         }));
     }
-    target_payable.oracle_approved = true;
-    payables_bucket.save(oracle_approval.payable_uuid.as_bytes(), &target_payable)?;
+    scope_attribute.oracle_approved = true;
+    // Add messages that will remove the current attribute and replace it with the attribute with an
+    // oracle approval on it
+    messages.append(
+        &mut provenance_util
+            .upsert_attribute_to_scope(&scope_attribute, &state.contract_name)?
+            .to_vec(),
+    );
     Ok(Response::new()
         .add_messages(messages)
-        .add_attribute(ORACLE_APPROVED_KEY, &target_payable.payable_uuid)
-        .add_attribute(PAYABLE_TYPE_KEY, state.payable_type)
-        .add_attribute(PAYABLE_UUID_KEY, target_payable.payable_uuid))
+        .add_attribute(ORACLE_APPROVED_KEY, &scope_attribute.payable_uuid)
+        .add_attribute(PAYABLE_TYPE_KEY, &scope_attribute.payable_type)
+        .add_attribute(PAYABLE_UUID_KEY, &scope_attribute.payable_uuid)
+        .add_attribute(ORACLE_ADDRESS_KEY, scope_attribute.oracle_address.as_str()))
```

Comment:
This is a very important addition - we have to broadcast the oracle address in every transaction (register payable, oracle approval, make payment) so that the oracle backend app can pick up only event stream attributes that link to its own address and ignore everything else.  Backend needs to be changed to match this new strat